UnityProject/Assets/Scripts/Objects/VisibleBehaviour.cs

```diff
@@ -93,7 +106,19 @@ private void UpdateState(bool _aliveState)
 
 		for (int i = 0; i < renderers.Length; i++)
 		{
-			renderers[i].enabled = _aliveState;
+			// Cast and check cast
+			SpriteRenderer sr = renderers[i] as SpriteRenderer;
+			if (sr != null)
+			{

```

Comment:
Is this needed? renderers should only have SpriteRenderers in them anyway

Reply:
I have to type cast to pass it to CanBeDisabled, and a few fail cast. So i'm keeping this bit unless you have another solution.